src/main/java/macrobase/analysis/pipeline/operator/MBOperator.java

```diff
@@ -1,3 +1,33 @@
 package macrobase.analysis.pipeline.operator;
 
-public interface MBOperator<S, T> extends MBConsumer<S>, MBProducer<T> { }
+import macrobase.analysis.pipeline.stream.MBStream;
+
+import java.util.List;
+
+public abstract class MBOperator<S, T> implements MBConsumer<S>, MBProducer<T> {
+    public <Y> MBOperator<S, Y> then(MBOperator<T, Y> o2, int batchSize) {
+        MBOperator<S, T> self = this;
+        return new MBOperator<S, Y>() {
+            @Override
+            public MBStream<Y> getStream() throws Exception {
+                return o2.getStream();
+            }
+
+            @Override
+            public void initialize() throws Exception { }
```

Comment:
If we want to actually use initialize in the future we probably want this chained operator to be like

```
public void initialize() throws Exception{
  self.initialize();
  o2.initialize();
}
```

and similarly with shutdown